packages/augur-core/source/contracts/0x/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -126,13 +126,16 @@ contract OrderValidationUtils is
             transferableTakerAssetAmount
         );
 
-        // Execute the maker transfers if a taker address is specified
-        if (order.takerAddress != address(0)) {
-            fillableTakerAssetAmount = getSimulatedOrderMakerTransferResults(
-                order,
-                order.takerAddress,
-                fillableTakerAssetAmount
-            ) == OrderTransferResults.TransfersSuccessful ? fillableTakerAssetAmount : 0;
+        if (!_isAssetDataValid(order.takerAssetData)) {
+            fillableTakerAssetAmount = 0;
+        }
+
+        if (order.takerFee != 0 && !_isAssetDataValid(order.takerFeeAssetData)) {
+            fillableTakerAssetAmount = 0;
+        }
+
+        if (orderInfo.orderStatus != LibOrder.OrderStatus.FILLABLE) {
+            fillableTakerAssetAmount = 0;
         }
```

Comment:
Any reason this isn't a single if-statement?

Reply:
just how the 0x contract is!